certora/specs/SocialRecoveryModule.spec

```diff
@@ -402,3 +403,15 @@ rule cancelRecoveryDoesNotAffectOtherWallet(env e, address otherWallet) {
         otherRequestBefore.newOwners[i] == otherRequestAfter.newOwners[i] &&
         otherWalletNonceBefore == currentContract.walletsNonces[otherWallet];
 }
+
+rule cantFinalizeRecoveryBeforeDelayPeriod(env e) {
+    requireInitiatedRecovery(safeContract);
+
+    uint64 recoveryTimestamp = currentContract.recoveryRequests[safeContract].executeAfter;
+
+    currentContract.finalizeRecovery@withrevert(e, safeContract);
+
+    bool success = !lastReverted;
+
+    assert success => require_uint64(e.block.timestamp) >= recoveryTimestamp, "Recovery finalized before delay period";
```

Comment:
TIL `require_uint64`

Where is this mentioned? I was not able to find it in the docs.

Reply:
I found this when reading openzeppelin contracts specifications. It seems to be mentioned in the CVL syntax under `require_functions` here: https://docs.certora.com/en/latest/docs/cvl/expr.html#syntax

Reply:
https://github.com/OpenZeppelin/openzeppelin-contracts/blob/dc625992575ecb3089acc35f5475bedfcb7e6be3/certora/specs/EnumerableSet.spec#L78

Reply:
I was trying to find the particular one (`require_uint64`) and that didn't bring any results in the docs. Thanks for the links!